dask/array/routines.py

```diff
@@ -161,6 +161,27 @@ def dot(a, b):
     return tensordot(a, b, axes=((a.ndim - 1,), (b.ndim - 2,)))
 
 
+@wraps(np.diff)
+def diff(a, n=1, axis=-1):
+    a = asarray(a)
+    n = int(n)
+    axis = int(axis)
+
+    axis %= a.ndim
```

Comment:
Change to avoid normalizing `axis`. With the change, we now catch out-of-bound errors when creating the slices themselves.